### DISCUSSION TITLE
events Click and wire:loading in layout

### DISCUSSION THREAD

**Discussion by kevin4dhd** [livewire/livewire#3210] [1. Help]
Hello, sorry for my bad English, it's me again, I have been reading all the documentation and I have not found the solution to this, also, I do not know if it is possible.

Let's imagine that I have the following layout.

![image](https://user-images.githubusercontent.com/22886367/123835954-c7177d00-d8ce-11eb-95a7-66f582435600.png)

If you notice, I want that in all my livewire pages, there is a button on top (it is an example, let's imagine that I want a loading bar, or anything else that is repeated in all the pages) and let's imagine that I want that button emit a click event, or a global event with emit, if it is not inside {{$ slot}} it doesn't work for me.

In summary, I want the events to work in the layout of all the livewire pages, if someone had an idea it would be of great help.

**Comment by ph7jack** [comment]
wire:loading only work inside a Livewire component.

**Comment by joshhanley** [comment]
@kevin4dhd as @PH7-Jack has said, you need to make sure any Livewire directives/attributes are actually inside a LW component or they won't work.

I've written about how to do what you after here, in a two part series:
Part 1: https://joshhanley.com.au/articles/how-to-use-layout-components-in-livewire
Part 2: https://joshhanley.com.au/articles/how-to-structure-your-layout-file-for-livewire

Hope this helps!

**Comment by kevin4dhd** [comment]
I have achieved it in the following way, in case someone wants it

![image](https://user-images.githubusercontent.com/22886367/123881008-6a858380-d909-11eb-9788-154f8ba045df.png)

and about the click event, I have just wrapped the code to be repeated in another component, then I call it with @livewire ('components.app-ecomerce-sidebar') and use $ emit to send the click event
